pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/StartServiceManagerCommand.java

```diff
@@ -218,11 +218,12 @@ private Map<String, Object> getDefaultConfig(ServiceRole serviceRole)
             ControllerConf.ControllerMode.DUAL, true);
       case BROKER:
         return PinotConfigUtils
-            .generateBrokerConf(_clusterName, _zkAddress, null, CommonConstants.Helix.DEFAULT_BROKER_QUERY_PORT);
+            .generateBrokerConf(_clusterName, _zkAddress, null, CommonConstants.Helix.DEFAULT_BROKER_QUERY_PORT, -1);
       case SERVER:
         return PinotConfigUtils
             .generateServerConf(_clusterName, _zkAddress, null, CommonConstants.Helix.DEFAULT_SERVER_NETTY_PORT,
-                CommonConstants.Server.DEFAULT_ADMIN_API_PORT, CommonConstants.Server.DEFAULT_GRPC_PORT, null, null);
+                CommonConstants.Server.DEFAULT_ADMIN_API_PORT, CommonConstants.Server.DEFAULT_GRPC_PORT, -1, -1, null,
```

Comment:
Why not just use the default port value just as the other configs do, so that you don't have to validate whether the value is `<= 0`?

Reply:
sounds good. i will make default port 0 in QueryConfig